sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -1401,6 +1401,40 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSparkSession {
     assert(e3.message.contains(errorMsg3))
   }
 
+  test("SPARK-33391: element_at with CreateArray") {
```

Comment:
It seems an overkill to have end-to-end test for it. How about we just add more tests in `CollectionExpressionsSuite.correctly handles ElementAt nullability for arrays` to test negative and invalid indices?

Reply:
sure

Reply:
updated